PWGLF/Tasks/cascadecorrelations.cxx

```diff
@@ -0,0 +1,234 @@
+
+      // PID
+      // Lambda check
+      if (casc.sign() < 0) {
+        // Proton check:
+        if (TMath::Abs(posTrack.tpcNSigmaPr()) > 3) {
+          cascflags(0);
+          continue;
+        }
+        // Pion check:
+        if (TMath::Abs(negTrack.tpcNSigmaPi()) > 3) {
+          cascflags(0);
+          continue;
+        }
+      } else {
+        // Proton check:
+        if (TMath::Abs(negTrack.tpcNSigmaPr()) > 3) {
+          cascflags(0);
+          continue;
+        }
+        // Pion check:
+        if (TMath::Abs(posTrack.tpcNSigmaPi()) > 3) {
+          cascflags(0);
+          continue;
+        }
+      }
+      // Bachelor check
+      if (TMath::Abs(bachTrack.tpcNSigmaPi()) < 3) {
+        if (TMath::Abs(bachTrack.tpcNSigmaKa()) < 3) {
+          // TODO: ambiguous! ignore for now
+          cascflags(0);
+          continue;
+        }
+        cascflags(1);
+        continue;
+      } else if (TMath::Abs(bachTrack.tpcNSigmaKa()) < 3) {
+        cascflags(2);
+        continue;
+      }
+      // if we reach here, the bachelor was neither pion nor kaon
+      cascflags(0);
+    } // cascade loop
+  }   // process
+};    // struct
+
+struct cascadeCorrelations {
+  HistogramRegistry registry{
+    "registry",
+    {
+      {"hMassXiMinus", "hMassXiMinus", {HistType::kTH1F, {{3000, 0.0f, 3.0f, "Inv. Mass (GeV/c^{2})"}}}},
+      {"hMassXiPlus", "hMassXiPlus", {HistType::kTH1F, {{3000, 0.0f, 3.0f, "Inv. Mass (GeV/c^{2})"}}}},
+      {"hMassOmegaMinus", "hMassOmegaMinus", {HistType::kTH1F, {{3000, 0.0f, 3.0f, "Inv. Mass (GeV/c^{2})"}}}},
+      {"hMassOmegaPlus", "hMassOmegaPlus", {HistType::kTH1F, {{3000, 0.0f, 3.0f, "Inv. Mass (GeV/c^{2})"}}}},
+      {"hPhi", "hPhi", {HistType::kTH1F, {{100, 0, 2 * PI, "#varphi"}}}},
+      {"hDeltaPhiSS", "hDeltaPhiSS", {HistType::kTH1F, {{100, -PI / 2, 1.5 * PI, "#Delta#varphi"}}}},
+      {"hDeltaPhiOS", "hDeltaPhiOS", {HistType::kTH1F, {{100, -PI / 2, 1.5 * PI, "#Delta#varphi"}}}},
+    },
+  };
+
+  Filter Selector = aod::cascadeflags::isSelected > 0; // TODO: treat Omega's and Xi's differently
+
+  void process(soa::Join<aod::Collisions, aod::EvSels>::iterator const& collision, soa::Filtered<aod::CascDataExtSelected> const& Cascades, aod::V0sLinked const&, aod::V0Datas const&, FullTracksExtIU const&)
+  {
+    // Some QA on the cascades
+    for (auto& casc : Cascades) {
+
+      auto v0 = casc.v0_as<o2::aod::V0sLinked>();
+      if (!(v0.has_v0Data())) {
+        continue; // reject if no v0data
+      }
+      auto v0data = v0.v0Data();
+
+      if (casc.sign() < 0) { // FIXME: could be done better...
+        registry.fill(HIST("hMassXiMinus"), casc.mXi());
+        registry.fill(HIST("hMassOmegaMinus"), casc.mOmega());
+      } else {
+        registry.fill(HIST("hMassXiPlus"), casc.mXi());
+        registry.fill(HIST("hMassOmegaPlus"), casc.mOmega());
+      }
+
+      registry.fill(HIST("hPhi"), RecoDecay::phi(casc.px(), casc.py()));
```

Comment:
To get the phi, you could use the method phi() implemented in the LFStrangenessTable (which by the way does exactly what you are doing)